scripts/plot_scalar_results.py

```diff
@@ -702,6 +702,74 @@ def subplot_flow_out_multi_carrier(carriers):
         except Exception as e:  # noqa 722
             logger.warning(f"Could not plot {output_path_plot}: {e}.")
 
+    def plot_demands_stacked_carriers(carriers):
+        carriers.append("ch4")
+        var_name = [f"flow_in_{carrier}" for carrier in carriers]
+        tech = "demand"
+        unit = "Wh"
+        output_path_plot = os.path.join(
+            target, "demand_stacked_carriers_" + "_".join(carriers) + ".png"
+        )
+        plot = ScalarPlot(scalars)
+        plot.select_data(var_name=var_name)
+        plot.selected_scalars = dp.filter_df(
+            plot.selected_scalars, column_name="tech", values=tech, inverse=False
+        )
+        plot.selected_scalars.replace({"flow_in_*": ""}, regex=True, inplace=True)
+        plot.prepared_scalar_data = aggregate_regions(plot.selected_scalars)
+
+        plot.prepared_scalar_data = plot.prepared_scalar_data.reset_index()
+        plot.prepared_scalar_data = plot.prepared_scalar_data.set_index(
+            ["scenario", "var_name"]
+        )
+
+        plot.prepared_scalar_data = plot.prepared_scalar_data.filter(
+            items=["var_value"]
+        )
+        plot.prepared_scalar_data = plot.prepared_scalar_data.unstack("var_name")
+
+        column_names = plot.prepared_scalar_data.columns
+        plot.prepared_scalar_data = plot.prepared_scalar_data.T.reset_index(drop=True).T
+
+        for column_num, column_name in enumerate(column_names):
+            plot.prepared_scalar_data.rename(
+                columns={column_num: column_name[1] + "-demand"}, inplace=True
+            )
+
```

Comment:
Please write some inline comments to organize and explain what is happening here.

Reply:
Done with commit https://github.com/rl-institut/oemof-B3/pull/228/commits/2546321792b80dcba16b80bcd4a0a7a3f62cd9be.